src/service.js

```diff
@@ -1,192 +1,357 @@
+    // (Mocha only)
+    if (test._retriedTest && passed) {
+      --this._failures;
+      return;
+    }
+
+    // don't bump failure number if test was retried and still failed
+    // (Mocha only)
+    if (
+      test._retriedTest &&
+      !passed &&
+      (
+        typeof test._currentRetry === 'number' &&
+        typeof test._retries === 'number' &&
+        test._currentRetry < test._retries
+      )
+    ) {
+      return;
+    }
+
+    const isJasminePendingError = typeof error === 'string' && error.includes('marked Pending');
+    if (!passed && !isJasminePendingError) {
+      ++this._failures;
+      this._failReasons.push((error && error.message) || 'Unknown Error')
     }
   }
 
-  afterScenario(world, { passed, error, duration }) {
-    console.log(error, duration)
-    if (!passed) {
-      ++this.failures;
+  afterScenario(world, result) {
+    const { passed } = result || {};
+    this._specsRan = true;
+    const status = world.result?.status.toLowerCase();
+    if (status !== 'skipped') {
+      this._scenariosThatRan.push(world.pickle.name || 'unknown pickle name');
+    }
+    if (status && this._failureStatuses.includes(status)) {
+      const exception = (
+        (world.result && world.result.message) ||
+        (status === 'pending'
+          ? `Some steps/hooks are pending for scenario "${world.pickle.name}"`
+          : 'Unknown Error'
+        )
+      )
+      ++this._failures;
+      this._failReasons.push(exception)
+    } else if (typeof passed !== 'undefined' && !passed) {
+      ++this._failures;
     }
   }
 
   after(result) {
-    if (!this.isServiceEnabled) {
+    if (!this._isServiceEnabled) {
       return;
     }
 
-    let failures = this.failures;
+    let failures = this._failures;
 
-    if (global.browser.options.mochaOpts && global.browser.options.mochaOpts.bail && Boolean(result)) {
+    // set _failures if user has bail option set in which case afterTest and
+    // afterSuite aren't executed before after hook
+    if (this._config.mochaOpts && this._config.mochaOpts.bail && Boolean(result)) {
       failures = 1;
     }
-    
+
     if (result === 0) {
       failures = 0;
     }
+
+    const { preferScenarioName } = this._options;
+    // For Cucumber: Checks scenarios that ran (i.e. not skipped) on the session
+    // Only 1 Scenario ran and option enabled => Redefine session name to Scenario's name
+    if (preferScenarioName && this._scenariosThatRan.length === 1) {
+      this._fullTitle = this._scenariosThatRan.pop();
+    }
+
     const status = 'status: ' + (failures > 0 ? 'failed' : 'passed');
 
-    if (!global.browser.isMultiremote) {
-      log.info(`Update job with sessionId ${global.browser.sessionId}, ${status}`);
-      return this._update(global.browser.sessionId, failures);
+    if (!this._browser.isMultiremote) {
+      log.info(`Update job with sessionId ${this._browser.sessionId}, ${status}`);
+      return this._update(this._browser.sessionId, failures);
```

Comment:
In certain cases if some unhandled exception occurs, after test hook is skipped and this._failures will not be set & in those cases correct status won't be reflected on the LT dashboard.